src/content/app/tools/blast/state/blast-api/blastApiSlice.ts

```diff
@@ -27,8 +43,45 @@ const blastApiSlice = restApiSlice.injectEndpoints({
         url: `${config.toolsApiBaseUrl}/blast/config`
       }),
       keepUnusedDataFor: 60 * 60 // one hour
+    }),
+    submitBlast: builder.mutation<
+      { submissionId: string; submission: BlastSubmission },
+      BlastSubmissionPayload
+    >({
+      query(payload) {
+        const body = {
+          genomeIds: payload.species.map(({ genome_id }) => genome_id),
+          querySequences: payload.querySequences,
+          parameters: payload.parameters
+        };
+
+        return {
+          url: `${config.toolsApiBaseUrl}/blast/job`,
+          method: 'POST',
+          body
+        };
+      },
+      transformResponse(response: BlastSubmissionResponse, _, payload) {
+        const { submissionId, jobs } = response;
+        // TODO: decide what to do when a submission returns error jobs
+        const results = jobs.map((job) => ({
+          jobId: job.jobId as string,
+          status: 'RUNNING',
+          seen: false,
+          data: null
+        }));
+        return {
+          submissionId,
+          submission: {
+            submission: payload,
+            results,
+            submittedAt: Date.now()
+          } as unknown as BlastSubmission
+        };
```

Comment:
@msmanoj you pointed out in https://github.com/Ensembl/ensembl-client/pull/719 that you don't like a `submission` field within another `submission` field.

In order to address this, we need to consider the `BlastSubmission` type, which is declared in `blastResultsSlice`:

https://github.com/Ensembl/ensembl-client/blob/6ec22fdea53e303ae768d300bc78252aabd9f406/src/content/app/tools/blast/state/blast-results/blastResultsSlice.ts#L37-L50

You will notice that this type, although called `BlastSubmission`, has a `submission` field inside of it. The purpose of this field is to keep the content of the blast submission form sent to the tools api. If you can think of another name, or of a different shape of this type, I am open for suggestions.

By the way, this type is likely to change at least somewhat after https://www.ebi.ac.uk/panda/jira/browse/ENSWBSITES-1566 is completed.

Reply:
Would it be okay rename the `submission` key on L75 to `submissionData`?

Reply:
Meh. It's not like the word `data` adds any extra meaning.

Reply:
Changed the field name to `submittedData`.